src/snowflake/snowpark/_internal/packaging_utils.py

```diff
@@ -0,0 +1,251 @@
+    for metadata_file in metadata_files:
+        parent_folder = os.path.dirname(metadata_file)
+        package = get_package_name_from_metadata(metadata_file)
+
+        if package is not None:
+            # Determine which folders or files belong to this package
+            record_file_path = os.path.join(parent_folder, "RECORD")
+            if os.path.exists(record_file_path):
+                # Get unique root folder names
+                with open(record_file_path, encoding="utf-8") as record_file:
+                    record_entries = list(
+                        {
+                            os.path.split(line)[0].split(",")[0]
+                            for line in record_file.readlines()
+                        },
+                    )
+                    included_record_entries = []
+                    for record_entry in record_entries:
+                        record_entry_full_path = os.path.abspath(
+                            os.path.join(directory, record_entry),
+                        )
+                        # RECORD file might contain relative paths to items outside target folder. (ignore these)
+                        if (
+                            os.path.exists(record_entry_full_path)
+                            and directory in record_entry_full_path
+                            and len(record_entry) > 0
+                        ):
+                            included_record_entries.append(record_entry)
+                    package_req = Requirement.parse(package)
+                    return_dict[package_req] = included_record_entries
+    return return_dict
+
+
+def identify_supported_packages(
+    packages: List[Requirement],
+    valid_packages: Dict[str, List[str]],
+    native_packages: Set[str],
+) -> Tuple[List[Requirement], List[Requirement], List[Requirement]]:
+    supported_dependencies: List = []
+    dropped_dependencies: List = []
+    new_dependencies: List = []
+    for package in packages:
+        package_name = package.name
+        package_version_req = package.specs[0][1] if package.specs else None
+        if package_name in valid_packages:
+            if (package_version_req is None) or (
+                package_version_req in valid_packages[package_name]
+            ):
+                supported_dependencies.append(package)
+            elif package_name in native_packages:
+                # Native packages should anaconda dependencies if possible, even if the version is not available
+                _logger.warning(
+                    f"Package {package_name}(version {package_version_req}) is an unavailable native "
+                    f"dependency, switching to latest available version "
+                    f"{valid_packages[package_name][-1]} instead."
+                )
+                dropped_dependencies.append(package)
+                new_dependencies.append(Requirement.parse(package_name))
+                native_packages.remove(package_name)
```

Comment:
Why do we remove it from `native_packages`?

Reply:
Because we have found a supported version of it on Anaconda, it is no longer problematic (native packages are a problem if they are unsupported only). The native_packages variable is used later to raise an error about unsupported native packages.